src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -13,67 +13,52 @@ import { ext } from "../../extensionVariables";
 import { getLocalSettingsJson, ILocalSettingsJson } from "../../funcConfig/local.settings";
 import { localize } from "../../localize";
 import * as api from '../../vscode-azurefunctions.api';
-import { confirmOverwriteSettings } from "./confirmOverwriteSettings";
 import { decryptLocalSettings } from "./decryptLocalSettings";
 import { encryptLocalSettings } from "./encryptLocalSettings";
+import { filterDownloadAppSettings } from "./filterDownloadAppSettings";
 import { getLocalSettingsFile } from "./getLocalSettingsFile";
 
-// Valen: Imported in registerCommands
 export async function downloadAppSettings(context: IActionContext, node?: AppSettingsTreeItem): Promise<void> {
-    // Valen: node is an optional parameter so this checks if it is null to set a value to it.
     if (!node) {
         node = await ext.tree.showTreeItemPicker<AppSettingsTreeItem>(AppSettingsTreeItem.contextValue, context);
     }
 
     const client: IAppSettingsClient = node.client;
-    //  Valen: runWithTemporaryDescription displays a 'Loading...' icon and temporarily changes context's
-    // description while `downloadAppSettingsInternal` is being run
     await node.runWithTemporaryDescription(context, localize('downloading', 'Downloading...'), async () => {
         await downloadAppSettingsInternal(context, client); // Valen: definition below
```

Comment:
so remove the reassignment above, and change this to `await downloadAppSettingsInternal(context, node.client); `

reassignments are great if being used multiple times below, otherwise we are adding more items on the stack.

Reply:
we can also remove the `// Valen: definition below` comment.

Reply:
Sorry about the comment I must've missed it but just removed it. I tried doing the fix you suggested and got the following error: 
<img width="398" alt="node client error" src="https://user-images.githubusercontent.com/70758549/125717614-0b7f6267-f7ba-47b2-b460-061853302ed7.png">
Any ideas as to why that may be?

Reply:
hmm.. that is interesting.. can the node.client be undefined by any chance or null? if should, you probably need to have check for that. It might be that you need to update downloadAppSettingsInternal to accept and `undefined` type and make check against that.

Reply:
if the user reached to the point where they right clicked on their function app to download app settings, that probably means they're signed in which would mean that node.client wouldn't be null correct?